modules/getintentBidAdapter.js

```diff
@@ -1,78 +1,130 @@
+  aliases: ['getintentAdapter'],
+  supportedMediaTypes: ['video', 'banner'],
 
-  function _callBids(params) {
-    if (typeof window.gi_hb === 'undefined') {
-      adloader.loadScript(headerBiddingStaticJS, function() {
-        bid(params);
-      }, true);
-    } else {
-      bid(params);
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid to validate.
+   * @return {boolean} True if this is a valid bid, and false otherwise.
+   * */
+  isBidRequestValid: function(bid) {
+    return !!(bid && bid.params && bid.params.pid && bid.params.tid);
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {BidRequest[]} bidRequests - an array of bids.
+   * @return ServerRequest[]
+   */
+  buildRequests: function(bidRequests) {
+    return bidRequests.map(bidRequest => {
+      let giBidRequest = buildGiBidRequest(bidRequest);
+      return {
+        method: 'GET',
+        url: buildUrl(giBidRequest),
+        data: giBidRequest,
+      };
+    });
+  },
+
+  /**
+   * Callback for bids, after the call to DSP completes.
+   * Parse the response from the server into a list of bids.
+   *
+   * @param {object} serverResponse A response from the server.
+   * @param {BidRequest} bidRequest
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, bidRequest) {
+    const bids = [];
+    if (serverResponse && serverResponse.no_bid !== 1) {
+      let size = parseSize(serverResponse.size);
+      let bid = {
+        requestId: bidRequest.bidId,
+        bidderCode: spec.code,
+        cpm: serverResponse.cpm,
+        width: size[0],
+        height: size[1],
+        mediaType: bidRequest.mediaType || 'banner'
+      };
+      if (bidRequest.creative_id) bid.creativeId = bidRequest.creative_id;
+      if (bidRequest.mediaType === 'video') {
+        bid.vastUrl = serverResponse.vast_url;
+        bid.descriptionUrl = serverResponse.vast_url;
+      } else {
+        bid.ad = serverResponse.ad;
+      }
+      bids.push(bid);
     }
+    return bids;
   }
 
-  function addOptional(params, request, props) {
-    for (var i = 0; i < props.length; i++) {
-      if (params.hasOwnProperty(props[i])) {
-        request[props[i]] = params[props[i]];
+}
+
+function buildUrl(bid) {
+  return '//' + BID_HOST + (bid.is_video ? BID_VIDEO_PATH : BID_BANNER_PATH);
+}
+
+/**
+ * Builds GI bid request from BidRequest.
+ *
+ * @param {BidRequest} bidRequest.
+ * @return {object} GI bid request.
+ * */
+function buildGiBidRequest(bidRequest) {
+  let giBidRequest = {
+    pid: bidRequest.params.pid, // required
+    tid: bidRequest.params.tid, // required
+    known: bidRequest.params.known || 1,
+    is_video: bidRequest.mediaType === 'video',
+    resp_type: 'JSON'
+  };
+  if (bidRequest.sizes) {
+    // TODO: add support for multiple sizes
+    giBidRequest.size = bidRequest.sizes[0].join('x');
```

Comment:
Adunit sizes may be defined as `[300, 250]`, in which case `sizes[0]` would be `300` and cause a `TypeError`. Be sure to check that `bidRequest.sizes[0]` is an array before calling `join`